zerolog/zerolog.go

```diff
@@ -0,0 +1,83 @@
+package zerolog
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/airbrake/gobrake/v5"
+	"github.com/buger/jsonparser"
+	"github.com/rs/zerolog"
+)
+
+type WriteCloser struct {
+	Gobrake *gobrake.Notifier
+}
+
+// Validates the WriteCloser matches the io.WriteCloser interface
+var _ io.WriteCloser = (*WriteCloser)(nil)
+
+// New creates a new WriteCloser
+func New(notifier *gobrake.Notifier) (io.WriteCloser, error) {
+	if notifier == nil {
+		return &WriteCloser{}, errors.New("airbrake notifier not provided")
+	}
+	return &WriteCloser{Gobrake: notifier}, nil
+}
+
+// Write parses the log data and sends off error notices to airbrake
+func (w *WriteCloser) Write(data []byte) (int, error) {
+	lvlStr, err := jsonparser.GetUnsafeString(data, zerolog.LevelFieldName)
+	if err != nil {
+		return 0, fmt.Errorf("error getting zerolog level: %w", err)
+	}
+
+	lvl, err := zerolog.ParseLevel(lvlStr)
+	if err != nil {
+		return 0, fmt.Errorf("error parsing zerolog level: %w", err)
+	}
+
+	if lvl < zerolog.ErrorLevel || lvl > zerolog.PanicLevel {
```

Comment:
I don't think the `fatal` level will work as proposed because fatal function calls the `os.Exit(1)` function.

Reply:
I'll test it out on my end. Good looking out.

Reply:
This is in fact correct, since os.Exit(1) is called the async process that gobrake provides does not have time to actually execute. Will revert to only checking the `Error` level